packages/hash/api/src/graphql/resolvers/paginationConnection/pageSearchResultConnection.ts

```diff
@@ -0,0 +1,137 @@
+import { ApolloError, UserInputError } from "apollo-server-express";
+import { EntitiesDocument } from "@hashintel/hash-backend-utils/search/doc-types";
+import { SearchHit } from "@hashintel/hash-backend-utils/search/adapter";
+
+import {
+  Resolver,
+  QueryPageSearchResultConnectionArgs,
+  PageSearchResultConnection,
+  PageSearchResult,
+} from "../../apiTypes.gen";
+import { GraphQLContext } from "../../context";
+
+// These values are the same defined in searchPages.
+// The name of the search index containing entities and the document field to perform
+// the search on. See the README for the `search-loader` for more details.
+const ENTITIES_SEARCH_INDEX = "entities";
+const ENTITIES_SEARCH_FIELD = "fullTextSearch";
+
+type TextSearchHit = Omit<SearchHit, "document"> & {
+  document: EntitiesDocument;
+};
+
+/**
+ * Paginated search over user pages.
+ * Currently doesn't support searching for both user and organization pages.
+ */
+export const pageSearchResultConnection: Resolver<
+  Promise<PageSearchResultConnection>,
+  {},
+  GraphQLContext,
+  QueryPageSearchResultConnectionArgs
+> = async (_, { accountId, query, pageSize, after }, ctx, __) => {
+  const { search, db } = ctx.dataSources;
+  if (!search) {
+    throw new ApolloError(
+      `Search is currently disabled so this search can't return any results`,
+      "INTERNAL_SERVER_ERROR",
+    );
+  }
+
+  if (query == null && after == null) {
```

Comment:
Would it be `(!query && !after)` instead? The suggested change would change semantics here if I'm not mistaken

Reply:
My mistake, I meant to suggest `(!query && !after)` here yes. The reason for this is because I wan't to avoid uses of `==` in a JS codebase where we primarily use `===`. I appreciate your usage here isn't wrong in any way, it's just harder to read what is really happening here in the context of codebase.